### PR TITLE
Make filter_record public to enable subclasses to override

### PR DESCRIPTION
**Pull request by frankier**
Subclasses are quite likely to want to override this if they want to provide custom fitlers, so let's make it public.

### CODE REVIEW

**Comment by frankier** [comment]
Consider this a first draft of the level of customization that I would like. I'm open to suggestions for other approaches. See here for how it was used to implement custom filtering: https://github.com/mood-mapping-muppets/repo/blob/2bc4f5e/ingress/covidmarch.py https://github.com/mood-mapping-muppets/repo/blob/2bc4f5e73c55adb6f35ce1dc848e39b58378bfbd/ingress/mynewsplease.py

File: newsplease/crawler/commoncrawl_crawler.py [hunk 1]
```diff
@@ -216,7 +216,8 @@ def __start_commoncrawl_extractor(warc_download_url, callback_on_article_extract
                                   log_level=logging.ERROR,
                                   delete_warc_after_extraction=True,
                                   continue_process=True,
-                                  log_pathname_fully_extracted_warcs=None):
+                                  log_pathname_fully_extracted_warcs=None,
+                                  extractor_cls=CommonCrawlExtractor):
```
Comment (fhamborg) [inline newsplease/crawler/commoncrawl_crawler.py:220]: Why do you think passing the CommonCrawlExtractor class here is necessary or what are the advantages of doing so compared to how it's currently implemented? Other than that, the PR is fine to me, i.e., making the filter method public.

**Review by fhamborg — COMMENTED** [review]
Thanks for the PR. Please see my question.

File: newsplease/crawler/commoncrawl_crawler.py (hunk 1 above)
Comment (frankier) [reply newsplease/crawler/commoncrawl_crawler.py:220]: The idea is that a user can subclass `CommonCrawlExtractor` themselves and then pass their subclass to `crawl_from_commoncrawl`. This way they can provide programmatic filters by overriding `filter_record(...)`.

Comment (fhamborg) [reply newsplease/crawler/commoncrawl_crawler.py:220]: This makes sense but I think this requires some brief explanation, e.g., as a comment perhaps at the following points: https://github.com/fhamborg/news-please/blob/563de2b7d7058cf89b894d581fea023a82b3f21a/newsplease/crawler/commoncrawl_crawler.py#L277 and here https://github.com/fhamborg/news-please/blob/563de2b7d7058cf89b894d581fea023a82b3f21a/newsplease/examples/commoncrawl.py

**Comment by frankier** [comment]
Done.

**Pull request merged**